crates/turbopack-dev/js/src/runtime.js

```diff
@@ -530,21 +531,26 @@ function runModuleExecutionHooks(module, executeModule) {
       ? globalThis.$RefreshInterceptModuleExecution$(module.id)
       : () => {};
 
-  executeModule({
-    register: globalThis.$RefreshReg$,
-    signature: globalThis.$RefreshSig$,
-  });
+  try {
+    executeModule({
+      register: globalThis.$RefreshReg$,
+      signature: globalThis.$RefreshSig$,
+    });
 
-  if ("$RefreshHelpers$" in globalThis) {
-    // This pattern can also be used to register the exports of
-    // a module with the React Refresh runtime.
-    registerExportsAndSetupBoundaryForReactRefresh(
-      module,
-      globalThis.$RefreshHelpers$
-    );
+    if ("$RefreshHelpers$" in globalThis) {
+      // This pattern can also be used to register the exports of
+      // a module with the React Refresh runtime.
+      registerExportsAndSetupBoundaryForReactRefresh(
+        module,
+        globalThis.$RefreshHelpers$
+      );
+    }
+  } catch (e) {
+    throw e;
+  } finally {
+    // Always cleanup the intercept, even if module execution failed.
+    cleanupReactRefreshIntercept();
```

Comment:
This change isn't related to the issue, but since we don't expect the runtime to completely crash and burn after a module instantiation error occurs, we should still clean up any side-effects.